frontend/src/app/auth/auth.interceptor.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpRequest,
+  HttpHandler,
+  HttpEvent,
+  HttpInterceptor,
+} from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { AuthService } from './services/auth.service';
+
+@Injectable()
+export class AuthInterceptor implements HttpInterceptor {
+  constructor(private authService: AuthService) {}
+
+  intercept(
+    request: HttpRequest<unknown>,
+    next: HttpHandler
+  ): Observable<HttpEvent<unknown>> {
+    if (this.authService.isLoggedIn()) {
+      const jwt = localStorage.getItem('access_token');
```

Comment:
It's better to create a method `getToken` in `authService` to encapsulate how and were you store the token